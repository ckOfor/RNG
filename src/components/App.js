// react
import React, { Component } from 'react';

// third party
import {
  Navbar,
  NavItem,
  Button,
  Row,
  Input,
  Dropdown,
  Modal,
  Table,
} from 'react-materialize'

// image
import logo from '../logo.svg';


// styles
import '../App.css';

// const fs = require('browserify-fs');
import fs from 'fs';

class App extends Component {
  state={
    generatedNumbers: [],
    userRange: '',
    minValue: '',
    maxValue: '',
    selected: '',
    mounted: false
  }
  
  componentDidMount(){
    this.mounted = true;
  }
  
  componentWillMount() {
    fs.readFile('../../data.txt', 'utf-8', (err, generatedNumbers) => !err
      ? this.setState({generatedNumbers: JSON.parse(generatedNumbers) }) : null); // eslint-disable-line
  }
  
  componentWillUnmount(){
    this.mounted = false;
  }
  
  onGenerateButtonClick = (userRange) => {
    return userRange > 0 && userRange <= 5000 &&
      this.phoneNumberGenerator(userRange)
  }
  
  phoneNumberGenerator = (userRange) => {
    let phoneNumbers, generatedNumbers = [];
    
    for ( let i = 0; i < userRange; i++) {
      phoneNumbers = `0${Math.floor(Math.random() * 1111111111)}`
      generatedNumbers.push({ id: i + 1,  value: phoneNumbers });
    }
    
    this.setState({ generatedNumbers })
    const dataToWrite = JSON.stringify(generatedNumbers);
    fs.writeFile('../../numbers.txt', dataToWrite, () => {});
    
    this.returnMaxAndMin(generatedNumbers)
  }
  
  sortGeneratedNumbers = (generatedNumbers, choice) => {
    if(choice === "asc") {
      this.setState({
        generatedNumbers: generatedNumbers.sort( function ( a, b )
        { return a.value - b.value; })
      })
      
      return generatedNumbers
    }
    
    if(choice === "dsc") {
      this.setState({
        generatedNumbers: generatedNumbers.sort( function ( a, b )
        { return b.value - a.value; })
      })
      
      return generatedNumbers
    }
  }
  
  returnMaxAndMin = (generatedNumbers, whatToReturn) => {
    let minValue = generatedNumbers[0].value;
    let maxValue = 0;
    
    Object.keys(generatedNumbers).forEach((value) => {
      if (generatedNumbers[value].value < minValue) {
        minValue = generatedNumbers[value].value
      }
    })
    
    Object.keys(generatedNumbers).forEach((value) => {
      if (maxValue < generatedNumbers[value].value) {
        maxValue = generatedNumbers[value].value
      }
    })
    
    this.setState({ minValue, maxValue })
  }
  
  render() {
    let { generatedNumbers } = this.state
    let num = [1,2,3,4,5]
    return (
      <div className="App">
        <Navbar brand='RNG' right>
          <NavItem>
            Total : { Object.keys(this.state.generatedNumbers).length && Object.keys(this.state.generatedNumbers).length }
          </NavItem>
        </Navbar>
        
        {/*<TableComponent list={this.state.generatedNumbers} />*/}
        <div style={{ marginLeft: 300, height: 500, width: 1000, marginTop: 50, display: 'block',
          overflow:'auto' }}>
          <Table className="table">
            <thead>
            <tr>
              <th data-field="id">S/N</th>
              <th data-field="name">Unique ID</th>
              <th data-field="price">Phone Numbers</th>
            </tr>
            </thead>
            <tbody>
            {
              (
                Object.keys(generatedNumbers).map((row, index) => {
                  return (
                    <tr key={index}>
                      <td>{index + 1}</td>
                      <td>{generatedNumbers[row].id}</td>
                      <td>{generatedNumbers[row].value}</td>
                    </tr>
                  )
                })
              )
            }
            </tbody>
          </Table>
        </div>
        <div style={{ marginLeft: 300, marginTop: 50 }}>
          <Row>
            {/*<Dropdown trigger={<Button style={{ marginTop: 20, marginLeft: 40  }}>Filter by:</Button>}>*/}
            {/*<NavItem*/}
            {/*className="asc"*/}
            {/*onClick={() => this.sortGeneratedNumbers(this.state.generatedNumbers, "asc")}>Ascending</NavItem>*/}
            {/*<NavItem*/}
            {/*className="dsc"*/}
            {/*onClick={() => this.sortGeneratedNumbers(this.state.generatedNumbers, "dsc")}>Descending</NavItem>*/}
            {/*<NavItem divider />*/}
            {/*<Modal*/}
            {/*header={`Minimum: ${this.state.minValue}`}*/}
            {/*trigger={<NavItem className="min">Min</NavItem>}>*/}
            {/*</Modal>*/}
            {/*<Modal*/}
            {/*header={`Maximum: ${this.state.maxValue}`}*/}
            {/*trigger={<NavItem className="max">Max</NavItem>}>*/}
            {/*</Modal>*/}
            {/*</Dropdown>*/}
            
            <Input
              className="range"
              s={2}
              label="Enter Range 1 <= 5000"
              onChange={userRange => this.setState({ userRange: userRange.target.value.replace(/\D/g, '') })}
              value={this.state.userRange}
            />
            <Button
              className="generate"
              style={{ marginTop: 20, marginLeft: 40 }}
              s={6}
              onClick={() => this.onGenerateButtonClick(this.state.userRange)}
              waves='light'>
              GENERATE
            </Button>
            
            <Button
              className="filter"
              style={{ marginTop: 20, marginLeft: 40 }}
              s={6}
              // onClick={() => this.onGenerateButtonClick(this.state.userRange)}
              waves='light'>
              filter
            </Button>
          </Row>
        </div>
      </div>
    );
  }
}

export default App;
