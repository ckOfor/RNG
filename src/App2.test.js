// react
import React from 'react';

// third-party
import ReactDOM from 'react-dom';
import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import { spy } from 'sinon';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

// component
import App from './components/App';

describe('<App />', () => {
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders three <Foo /> components', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('div')).to.have.lengthOf(8);
  });
  
  it('allows us to set props', () => {
    const wrapper = mount(<App bar="baz" />);
    expect(wrapper.props().bar).to.equal('baz');
    wrapper.setProps({ bar: 'foo' });
    expect(wrapper.props().bar).to.equal('foo');
  });
  
  it('calls componentDWillMount', () => {
    spy(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    expect(App.prototype.componentDidMount).to.have.property('callCount', 1);
    App.prototype.componentDidMount.restore();
  });
  
  it('renders three `.foo-bar`s', () => {
    const wrapper = render(<App />);
    expect(wrapper.find('div')).to.have.lengthOf(7);
  });
  
});
