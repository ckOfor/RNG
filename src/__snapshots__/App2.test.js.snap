// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App /> functions renders without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <Navbar
      brand="RNG"
      href="/"
      options={Object {}}
      right={true}
    >
      <nav>
        <div
          className="nav-wrapper"
        >
          <Col
            s={12}
          >
            <div
              className="col s12"
            >
              <a
                className="brand-logo"
                href="/"
              >
                RNG
              </a>
              <ul
                className="right hide-on-med-and-down"
              >
                <NavItem>
                  <li>
                    <a
                      href=""
                    >
                      Total : 
                      0
                    </a>
                  </li>
                </NavItem>
              </ul>
              <ul
                className="side-nav"
                id="nav-mobile"
              >
                <NavItem>
                  <li>
                    <a
                      href=""
                    >
                      Total : 
                      0
                    </a>
                  </li>
                </NavItem>
              </ul>
              <a
                className="button-collapse"
                data-activates="nav-mobile"
                href="#"
              >
                <Icon>
                  <i
                    className="material-icons"
                  >
                    view_headline
                  </i>
                </Icon>
              </a>
            </div>
          </Col>
        </div>
      </nav>
    </Navbar>
    <TableComponent
      list={Array []}
    >
      <div
        style={
          Object {
            "display": "block",
            "height": 500,
            "marginLeft": 300,
            "marginTop": 50,
            "overflow": "auto",
            "width": 1000,
          }
        }
      >
        <Table
          style={Object {}}
        >
          <table
            className=""
            style={Object {}}
          >
            <thead>
              <tr>
                <th
                  data-field="id"
                >
                  S/N
                </th>
                <th
                  data-field="name"
                >
                  Unique ID
                </th>
                <th
                  data-field="price"
                >
                  Phone Numbers
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </Table>
      </div>
    </TableComponent>
    <div
      style={
        Object {
          "marginLeft": 300,
          "marginTop": 50,
        }
      }
    >
      <Row>
        <div
          className="row"
        >
          <Input
            checked={false}
            label="Enter Range"
            onChange={[Function]}
            s={2}
            type="text"
            value=""
          >
            <div
              className="col input-field s2"
            >
              <input
                checked={false}
                className=""
                id="input_1"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className=""
                htmlFor="input_1"
              >
                Enter Range
              </label>
            </div>
          </Input>
          <Button
            className="generate"
            node="button"
            onClick={[Function]}
            s={6}
            style={
              Object {
                "marginLeft": 40,
                "marginTop": 20,
              }
            }
            waves="light"
          >
            <button
              className="btn waves-effect waves-light generate"
              disabled={false}
              onClick={[Function]}
              s={6}
              style={
                Object {
                  "marginLeft": 40,
                  "marginTop": 20,
                }
              }
            >
              GENERATE
            </button>
          </Button>
          <Button
            node="button"
            onClick={[Function]}
            s={6}
            style={
              Object {
                "marginLeft": 40,
                "marginTop": 20,
              }
            }
            waves="light"
          >
            <button
              className="btn waves-effect waves-light"
              disabled={false}
              onClick={[Function]}
              s={6}
              style={
                Object {
                  "marginLeft": 40,
                  "marginTop": 20,
                }
              }
            >
              Download
            </button>
          </Button>
        </div>
      </Row>
    </div>
  </div>
</App>
`;
